### PR TITLE
Feat: Update user profile #215

### PR DESCRIPTION
**Pull request by unyimeudemy**
# How should this be manually tested

-  Ensure to be a logged in user because all endpoints are protected.

- **Send a PATCH request to  `/api/v1/profile/{user_id}`** to update a user's profile based on the provided details in `UpdateUserProfileDto`.
  - **Success Response (200 OK)**: Returns `ProfileUpdateResponseDto` with updated user's profile.
  - **Error Responses**:
    - **404 Not Found**: If user is not found.
    - **401 Unauthorized**: For user not authorized.
    - **500 Internal server error**: For unexpected errors.

## Checklist
- [x] Implemented the Update User Profile endpoint to allow users to update their profile information.
- [x] Handled authentication and authorization issues.
- [x] Ensured proper input validation for the fields to be updated.
- [x] Wrote tests to cover the new functionality.
- [x] Ensured logging for error cases and successful updates.

## Reference Issue
- [[FEAT] : Update user profile #215](https://github.com/hngprojects/hng_boilerplate_java_web/issues/215)

### CODE REVIEW

File: src/main/java/hng_java_boilerplate/profile/entity/Profile.java [hunk 1]
```diff
@@ -1,24 +1,54 @@
...
+    @JsonProperty("department")
+    private String department;
+
+    @JsonProperty("social")
+    private String social;
+
+    @JsonProperty("bio")
+    private String bio;
+
+    @JsonProperty("recovery_email")
```
Comment (lurldgbodex) [inline src/main/java/hng_java_boilerplate/profile/entity/Profile.java:49]: is the recovery email suppose to be different from the user registered email?

File: src/main/resources/db/migration/V1.2_alter_profile_table.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
+ALTER TABLE profiles
+ADD COLUMN pronouns VARCHAR(255),
```
Comment (lurldgbodex) [inline src/main/resources/db/migration/V1.2_alter_profile_table.sql:2]: update version no to most recent

**Comment by unyimeudemy** [comment]
@lurldgbodex  all changes have been implemented

File: src/main/resources/db/migration/V1.2_alter_profile_table.sql (hunk 2 above)
Comment (lurldgbodex) [reply src/main/resources/db/migration/V1.2_alter_profile_table.sql]: The version has changed. you need to update again

Comment (lurldgbodex) [reply src/main/resources/db/migration/V1.2_alter_profile_table.sql]: and also resolve the conflict in the globalexceptionhandler

**Review by Fifetoyi — APPROVED** [review]
lgtm

**Pull request merged**